### ISSUE TITLE
1.9.0 soapClient not initialising

### ISSUE THREAD

**Issue by georgecookeIW** [ballerina-platform/module-ballerina-soap#75]
**Description:**
The new release of Ballerina/SOAP 1.9.0 breaks while initializing the SOAP Client. However, the same was supported by the earlier release of Ballerina/SOAP 1.0.0.
 
Given below is the system on which I experienced this issue.
OS: Windows 10, MacOS 10.14
Ballerina version – 1.2.4
 
Attached is the ballerina code. On building the attached file following error is thrown –
“ballerina build .\main.bal. It is a simple main function that tries to initialize the soap client.

```
import ballerina/io;
import ballerina/soap;

public function main() {
    io:println("Hello World!");
 
   soap:Soap12Client soap12Client = new("http://www.dneonline.com/calculator.asmx");
   xml body = xml `<m0:Add   xmlns:m0="http://tempuri.org">
                    <m0:intA>1</m0:intA>
                    <m0:intB>1</m0:intB>
                </m0:Add  >`;
    var response = soap12Client->sendReceive(<@untainted>body);
    io:println(response);
}
```

The same code builds without any error by changing the dependency of the ballerina/soap module to 1.0.0 in Ballerina.toml.

```
$ ballerina dist use jballerina-1.2.4
'jballerina-1.2.4' is the current active distribution version

$ ballerina run test
Compiling source
        georgecooke/test:0.1.0
error: ballerina/soap:1.9.0::/src/soap/soap11_endpoint.bal:39:42: variable 'soap11Client' is not initialized
error: ballerina/soap:1.9.0::/src/soap/soap11_endpoint.bal:50:41: variable 'soap11Client' is not initialized
error: ballerina/soap:1.9.0::/src/soap/soap11_endpoint.bal:60:32: variable 'soap11Client' is not initialized
error: ballerina/soap:1.9.0::/src/soap/soap12_endpoint.bal:39:42: variable 'soap12Client' is not initialized
error: ballerina/soap:1.9.0::/src/soap/soap12_endpoint.bal:50:41: variable 'soap12Client' is not initialized
error: ballerina/soap:1.9.0::/src/soap/soap12_endpoint.bal:60:32: variable 'soap12Client' is not initialized
error: georgecooke/test:0.1.0::main.bal:11:37: too many arguments in call to 'null()'
error: compilation contains errors
```

A quick look at your source code / commits show that the signature of the init() function has changed between 1.0.0 and 1.9.0; this is possibly the issue.

**Suggested Labels:**
bug

**Suggested Assignees:**
N/A

**Affected Product Version:**
Ballerina 1.2.4
module-ballerina-soap: 1.9.0

**OS, DB, other environment details and versions:**    
OS: Windows 10, MacOS 10.14
jBallerina: 1.2.4
(tested on 1.2.1 as well, but same result)

**Steps to reproduce:**
Ballerina run the above code in a project.

**Related Issues:**
N/A

**Comment by Bhashinee** [comment]
module-ballerina-soap: 1.9.0 is compatible with Ballerina Swan Lake - Preview 1 release. 
ballerina/soap version 1.0.0 is the latest soap module version supported with Ballerina 1.2.4.

You can import the 1.0.0 version like this,

`import ballerina/soap version 1.0.0;`

**Comment by georgecookeIW** [comment]
@Bhashinee thanks for your reply.

Brilliant - just what we needed as using the Ballerina TOML is not appropriate for the task at hand. It is not mentioned in any Ballerina documentation that this is possible with imports. This is a super simple, clear way to define / lock the version within the Ballerina code itself; I hope this can be documented clearly somewhere...

I mentioned this issue on Ballerina slack as well, and they mention that 1.9.0 Soap connector is only useable with the new "Ballerina Swan Lake" release (to be released) and is not backward compatible with 1.2.4. Again, not mentioned in documentation anywhere which is why we raised the issue...

Considering its not backward compatible with 1.2.4 (current latest Ballerina) there should be some way / method / process by which users can be informed during compilation that they are importing a module with no support for the locally installed Ballerina distribution.

Anyway, rant over; your fix works well - much appreciated.
